### PR TITLE
[netcore] Clear system last error before P/Invoke calls with SetLastError=true

### LINKED ISSUE #15691: Clear last error for SetLastError=true P/Invoke
CoreCLR always clears the last error for P/Invoke with SetLastError=true (https://github.com/dotnet/coreclr/pull/614). This is essential for APIs like `GetSysColor` where the failure cannot be detected from the return value alone (ie. failure and black color are indistinguishable). Another such example is `CreateMutex` and checking for `ERROR_ALREADY_EXISTS` last error which is set on success.

### PR DESCRIPTION
**Pull request by filipnavara**
Fixes #15691

### CODE REVIEW

**Comment by filipnavara** [comment]
@monojenkins build failed

File: mono/metadata/jit-icall-reg.h [hunk 1]
```diff
@@ -234,6 +234,7 @@ MONO_JIT_ICALL (mono_llvmonly_init_delegate) \
 MONO_JIT_ICALL (mono_llvmonly_init_delegate_virtual) \
 MONO_JIT_ICALL (mono_marshal_asany) \
 MONO_JIT_ICALL (mono_marshal_check_domain_image) \
+MONO_JIT_ICALL (mono_marshal_clear_last_error) \
 MONO_JIT_ICALL (mono_marshal_free) \
```
Comment (vargaz) [inline mono/metadata/jit-icall-reg.h]: Need to bump the aot image version in aot-runtime.h.

File: mono/mini/method-to-ir.c [hunk 2]
```diff
@@ -7386,6 +7386,12 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				}
 			}
 
+#ifdef ENABLE_NETCORE
+			if (save_last_error) {
+				mono_emit_jit_icall (cfg, mono_marshal_clear_last_error, NULL);
```
Comment (vargaz) [inline mono/mini/method-to-ir.c]: Extra {}

**Pull request merged**